File: backend/src/main/java/com/snackoverflow/toolgether/domain/postimage/service/PostImageService.java
package com.snackoverflow.toolgether.domain.postimage.service;

import com.snackoverflow.toolgether.domain.postimage.entity.PostImage;
import com.snackoverflow.toolgether.domain.postimage.repository.PostImageRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Service
@RequiredArgsConstructor
@Transactional(readOnly = true)
public class PostImageService {
    private final PostImageRepository postImageRepository;

    public List<PostImage> getPostImagesByPostId(Long postId) {
        return postImageRepository.findByPostId(postId);
    }
}

File: backend/src/main/java/com/snackoverflow/toolgether/domain/postimage/service/PostImageService.kt
package com.snackoverflow.toolgether.domain.postimage.service

import com.snackoverflow.toolgether.domain.postimage.entity.PostImage
import com.snackoverflow.toolgether.domain.postimage.repository.PostImageRepository
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
@Transactional(readOnly = true)
class PostImageService(
    val postImageRepository: PostImageRepository
) {
    fun getPostImagesByPostId(postId: Long): List<PostImage> {
        return postImageRepository.findByPostId(postId)
    }
}
